include/nbl/video/IGPUCommandBuffer.h

```diff
@@ -227,6 +227,24 @@ class NBL_API IGPUCommandBuffer :
             for (uint32_t i = _first + 1u; i < IGPUPipelineLayout::DESCRIPTOR_SET_COUNT; ++i)
                 _destPplnLayouts[i] = nullptr;
     }
+
+    uint32_t m_resetCount = 0;
+
+    virtual bool resetCommon()
+    {
+        m_resetCount = m_cmdpool->getResetCounter().load();
+        return base_t::reset(ERF_NONE);
+    }
+
+    // This should be called on every command buffer method
+    // Only doing it in begin for now, we expect that the command pool won't be reset midway through recording a command buffer
+    inline bool checkForCommandPoolReset()
+    {
+        if (m_cmdpool->getResetCounter().load() <= m_resetCount)
+            return false;
+
+        return resetCommon();
```

Comment:
`check` shouldn't call the reset_impl (or reset-common) since the only thing it needs to do is just transition the state into INITIAL

P.S. on another note it was a mistake to have a non-pure virtual `reset` in `ICommandBuffer`, unecessary complexity and stuff.

Reply:
Resolved.